client/src/components/AuthForm/SignUpForm/SignUpForm.js

```diff
@@ -39,15 +39,18 @@ const SignUpForm = ({
     return result.length < 4 || signUpForm.password.length < 6;
   };
 
-  const handleSubmit = async (e) => {
+  const handleSubmit = (e) => {
     e.preventDefault();
-    try {
-      const result = await register(signUpForm);
-      if (alert.length > 0) {
-        console.log(alert[0].message);
-      }
-    } catch (err) {
-      console.log(err);
+    if (!validator.isEmail(signUpForm.email)) {
+      dispatch(setAlert('Enter a valid email address.', 'Error'));
+    } else if (signUpForm.password.length < 6 || signUpForm.password === '') {
+      dispatch(setAlert('Create a password at least 6 characters long.'));
+    } else if (signUpForm.fullName === '' || signUpForm.username === '') {
```

Comment:
an empty string is just a falsy value inside an if statement

so convert all 

`foo.bar === ''
`
to 

`!foo.bar`